lib/taskana-core/src/main/java/pro/taskana/mappings/WorkbasketMapper.java

```diff
@@ -148,14 +150,24 @@
+    @Insert("<script>INSERT INTO WORKBASKET (ID, KEY, CREATED, MODIFIED, NAME, DOMAIN, TYPE, DESCRIPTION, OWNER, CUSTOM_1, CUSTOM_2, CUSTOM_3, CUSTOM_4, ORG_LEVEL_1, ORG_LEVEL_2, ORG_LEVEL_3, ORG_LEVEL_4, MARKED_FOR_DELETION) VALUES (#{workbasket.id}, #{workbasket.key}, #{workbasket.created}, #{workbasket.modified}, #{workbasket.name}, #{workbasket.domain}, #{workbasket.type}, #{workbasket.description}, #{workbasket.owner}, #{workbasket.custom1}, #{workbasket.custom2}, #{workbasket.custom3}, #{workbasket.custom4}, #{workbasket.orgLevel1}, #{workbasket.orgLevel2}, #{workbasket.orgLevel3}, #{workbasket.orgLevel4}, #{workbasket.markedForDeletion}) "
         + "</script>")
     @Options(keyProperty = "id", keyColumn = "ID")
     void insert(@Param("workbasket") WorkbasketImpl workbasket);
 
-    @Update("UPDATE WORKBASKET SET MODIFIED = #{workbasket.modified}, KEY = #{workbasket.key}, NAME = #{workbasket.name}, DOMAIN = #{workbasket.domain}, TYPE = #{workbasket.type}, DESCRIPTION = #{workbasket.description}, OWNER = #{workbasket.owner}, CUSTOM_1 = #{workbasket.custom1}, CUSTOM_2 = #{workbasket.custom2}, CUSTOM_3 = #{workbasket.custom3}, CUSTOM_4 = #{workbasket.custom4}, ORG_LEVEL_1 = #{workbasket.orgLevel1}, ORG_LEVEL_2 = #{workbasket.orgLevel2}, ORG_LEVEL_3 = #{workbasket.orgLevel3}, ORG_LEVEL_4 = #{workbasket.orgLevel4}  WHERE id = #{workbasket.id}")
+    @Update("UPDATE WORKBASKET SET MODIFIED = #{workbasket.modified}, KEY = #{workbasket.key}, NAME = #{workbasket.name}, DOMAIN = #{workbasket.domain}, TYPE = #{workbasket.type}, DESCRIPTION = #{workbasket.description}, OWNER = #{workbasket.owner}, CUSTOM_1 = #{workbasket.custom1}, CUSTOM_2 = #{workbasket.custom2}, CUSTOM_3 = #{workbasket.custom3}, CUSTOM_4 = #{workbasket.custom4}, ORG_LEVEL_1 = #{workbasket.orgLevel1}, ORG_LEVEL_2 = #{workbasket.orgLevel2}, ORG_LEVEL_3 = #{workbasket.orgLevel3}, ORG_LEVEL_4 = #{workbasket.orgLevel4}, MARKED_FOR_DELETION = #{workbasket.markedForDeletion} WHERE id = #{workbasket.id}")
     void update(@Param("workbasket") WorkbasketImpl workbasket);
 
+    @Update("UPDATE WORKBASKET SET MARKED_FOR_DELETION = #{markedForDeletion} WHERE id = #{workbasketId}")
+    void updateMarkedForDeletion(@Param("workbasketId") String workbasketId, @Param("markedForDeletion") boolean markedForDeletion);
+
+    @Select("<script>SELECT ID, MARKED_FOR_DELETION FROM WORKBASKET WHERE ID = #{workbasketId} "
+            + "<if test=\"_databaseId == 'db2'\">with UR </if> "
+            + "</script>")
+    @Results(value = {
+            @Result(property = "markedForDeletion", column = "MARKED_FOR_DELETION")})
+    boolean findMarkedForDeletionById(@Param("workbasketId") String workbasketId);
```

Comment:
same here. just Workbasket.get()?

Reply:
I remove it and use the existent findById method

Reply:
see above

Reply:
Done